### PR TITLE
GUACAMOLE-1846: Skip user promotion entirely if there are no pending users.

### PR DESCRIPTION
**Pull request by jmuehlner**

### CODE REVIEW

**Comment by jmuehlner** [comment]
This addresses an issue where unneeded display flushes would occur even when no users are being promoted.

File: src/libguac/client.c [hunk 1]
```diff
@@ -185,6 +185,12 @@ static void guac_client_promote_pending_users(union sigval data) {
     /* Acquire the lock for reading and modifying the list of pending users */
     guac_rwlock_acquire_write_lock(&(client->__pending_users_lock));
 
+    /* Skip user promotion entirely if there's no pending users */
+    if (client->__pending_users == NULL) {
+        guac_rwlock_release_lock(&(client->__pending_users_lock));
+        goto promotion_complete;
```
Comment (mike-jumper) [inline src/libguac/client.c:191]: To be more DRY, should the same be done for:

https://github.com/apache/guacamole-server/blob/974816a5caae45f5ecde5b4f2b258b9a513f2c06/src/libguac/client.c#L200-L205

File: src/libguac/client.c [hunk 2]
```diff
@@ -185,6 +185,12 @@ static void guac_client_promote_pending_users(union sigval data) {
     /* Acquire the lock for reading and modifying the list of pending users */
     guac_rwlock_acquire_write_lock(&(client->__pending_users_lock));
 
+    /* Skip user promotion entirely if there's no pending users */
+    if (client->__pending_users == NULL) {
+        guac_rwlock_release_lock(&(client->__pending_users_lock));
```
Comment (mike-jumper) [inline src/libguac/client.c:190]: Any reason to not move the `promotion_complete` label up a line and rely on the `guac_rwlock_release_lock()` call that already exists there?

**Review by mike-jumper — CHANGES_REQUESTED** [review]
As with the other GUACAMOLE-1846 changes, this should probably also be against `staging/1.5.4`.

**Comment by necouchman** [comment]
@jmuehlner It looks like #457 went into `staging/1.5.4`, so my guess is this should be based against that?

**Comment by mike-jumper** [comment]
Wow ... reviewed at nearly _exactly_ the same time.

File: src/libguac/client.c (hunk 1 above)
Comment (jmuehlner) [reply src/libguac/client.c:190]: Good call - it's done.

File: src/libguac/client.c (hunk 2 above)
Comment (jmuehlner) [reply src/libguac/client.c]: Nope!

**Review by mike-jumper — APPROVED** [review]

**Pull request merged**